src/CohortManager.js

```
@@ -0,0 +1,107 @@
+/* eslint-disable array-callback-return */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-unused-vars */
+const Cohort = require('./Cohort')
+const Student = require('./Student')
+
+class CohortManager {
+  constructor() {
+    this.cohorts = []
+    this.students = []
+    this.studentsId = 1
+  }
+
+  addCohort(name) {
+    const regex = /^[A-Za-z0-9 ]*$/
+    if (name === '') {
+      return 'invalid name'
+    }
+    if (regex.test(name)) {
+      const newCohort = new Cohort(name)
+      const selectedName = []
+
+      this.cohorts.push(newCohort)
+
+      this.cohorts.find((element) => {
+        if (element.name === name) {
+          selectedName.push(element)
+        }
+      })
+      return selectedName.length ? 'cohort already exist' : this.cohorts
```

```js
const cohortExists = this.cohorts.find((element) => element.name === name)

return cohortExists ? 'cohort already exist' : this.cohorts
```

You can two line this :)

But keep in mind my above comment about preferring throwing errors instead of returning strings